app/controllers/admin/reports_controller.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Admin
+  class ReportsController < Admin::ApplicationController
+    def index
+    end
+
+    def total_volunteer_minutes_by_office
+      respond_to do |format|
+        format.html { send_data generate_csv(%w(office_id volunteer_minutes), Office.total_volunteer_minutes_by_office), filename: "#{DateTime.current.to_i}-total-volunteer-minutes-by-office.csv"}
+        format.json { render json: Office.total_volunteer_minutes_by_office.to_json }
+      end
+    end
+
+    def total_volunteer_minutes_by_state
+      respond_to do |format|
+        format.html { send_data generate_csv(%w(state volunteer_minutes), Office.total_volunteer_minutes_by_state), filename: "#{DateTime.current.to_i}-total-volunteer-minutes-by-state.csv"}
+        format.json { render json: Office.total_volunteer_minutes_by_state.to_json }
+      end
+    end
+
+    def total_volunteer_minutes_by_county
+      respond_to do |format|
+        format.html { send_data generate_csv(%w(county volunteer_minutes), Office.total_volunteer_minutes_by_county(params['state'])), filename: "#{DateTime.current.to_i}-total-volunteer-minutes-by-county.csv"}
+        format.json { render json: Office.total_volunteer_minutes_by_county(params['state']).to_json }
+      end
+    end
+
+    def total_children_served_by_office
+      respond_to do |format|
+        format.html { send_data generate_csv(%w(office_id children_served), Office.total_children_served_by_office), filename: "#{DateTime.current.to_i}-total-children-served-by-office.csv"}
+        format.json { render json: Office.total_children_served_by_office.to_json }
+      end
+    end
+
+    def total_children_served_by_state
```

Comment:
could DRY these up a bit by doing something like:

```
TOTAL_CHILDREN_SERVED_BY_STATE = %w(state children_served).freeze

def total_children_served_by_state
  data  = Office.total_children_served_by_state

  respond_to do |format|
    format.html do
      send_data generate_csv(TOTAL_CHILDREN_SERVED_BY_STATE, data), 
                filename: filename('total-children-served-by-state')
    end
    format.json { render json: data.to_json }
  end
end

def filename(data_type)
  "#{DateTime.current.to_i}-#{data_type}.csv"
end
```

Not a deal breaker, just some ideas. Also, wondering why you're using format.html to return a csv? Maybe that will become obvious to me further down.

Reply:
I shouldn't be sending a csv via html. I just need to return a csv. I'll fix this approach.

Reply:
I didn't get a chance to look over this again. Happy to take another gander once you are feeling good about it.